### PR TITLE
styleguide modals functional

### LINKED ISSUE #893: Styleguide: Modals
Document Modals used within the app in a styleguide section

Includes Modals and Confirmation Modals
If any types are missing, please comment with example.

**Refactor duplicated modals where necessary.**

Add:

- `<Section name="modals" title="Modals"></Section>`
- define a `Modals.scss`
- define a `Modals.js` in `/styleguide/components/sections`

Styleguide Index: #884
reference #877

### PR DESCRIPTION
**Pull request by abemassry**
closes #893

### CODE REVIEW

File: src/styleguide/components/sections/Modals.js [hunk 1]
```diff
@@ -1,11 +1,121 @@
...
+          </div>
+        </ModalShell>
+        <br />
+        You can have form elements or other elements inside a general modal.
+      </StyleguideSection>
+    );
+  }
 }
+
+Modals.propTypes = {
```
Comment (mparke) [inline src/styleguide/components/sections/Modals.js]: empty propTypes

File: src/styleguide/components/sections/Modals.js [hunk 2]
```diff
@@ -1,11 +1,121 @@
...
+        <br />
+        <a onClick={() => this.setState({ confirmOpen: true })} className="btn btn-default">
+          Confirm modal
+        </a>
+        <ModalShell
+          title="Modal example"
+          open={this.state.confirmOpen}
+          close={() => { this.setState({ confirmOpen: false }); }}
+        >
+          <div>
```
Comment (mparke) [inline src/styleguide/components/sections/Modals.js]: you could also render the ConfirmModalBody in here, re-using the defaults already in that module.

File: src/layouts/Layout.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { eventRead, eventSeen } from '~/api/events';
 import Header from '~/components/Header';
 import Sidebar from '~/components/Sidebar';
 import Notifications from '~/components/Notifications';
-import Modal from './Modal';
+import ModalShell from './ModalShell';
```
Comment (mparke) [inline src/layouts/Layout.js]: can we put ModalShell back into `src/components/modals/` along with `ConfirmModalBody` so we know where to find these components?

Comment (mparke) [inline src/layouts/Layout.js]: we talked in person ( going to make a pull against this branch )

**Comment by eatonphil** [comment]
Is it within the scope of this PR to fix the broken reset root password modal? Pulled these changes and it still appears to be a problem.

**Comment by eatonphil** [comment]
Whoops, my bad. This does fix the reset password modal bug. Thanks!

File: src/layouts/Layout.js [hunk 4]
```diff
@@ -187,7 +187,13 @@ export class Layout extends Component {
...
       <div className="layout full-height">
         <PreloadIndicator />
-        <Modal />
+        <ModalShell
+          open={this.props.modal.open}
+          title={this.props.modal.title}
+          close={() => dispatch(hideModal())}
+        >
+          {this.props.modal.body}
+        </ModalShell>
```
Comment (eatonphil) [inline src/layouts/Layout.js]: +1

File: src/styleguide/components/sections/Modals.js [hunk 5]
```diff
@@ -1,11 +1,118 @@
...
+              <button
+                className="btn btn-default"
+                onClick={() => { this.setState({ generalOpen: false }); }}
+              >
+                Change
+              </button>
+            </div>
+          </div>
+        </ModalShell>
+        <br />
```
Comment (eatonphil) [inline src/styleguide/components/sections/Modals.js]: !!!

File: src/styleguide/components/sections/Modals.js [hunk 6]
```diff
@@ -1,11 +1,118 @@
...
+              >
+                OK
+              </button>
+            </div>
+          </div>
+        </ModalShell>
+        <br />
+        <br />
+        This is an example of a
+        <br />
```
Comment (eatonphil) [inline src/styleguide/components/sections/Modals.js]: !!!!

**Comment by briansteffens** [comment]
Should `test/components/ConfirmModal.spec.js` rename to `test/components/ConfirmModalBody.spec.js` ?

File: test/components/ConfirmModal.spec.js [hunk 7]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import sinon from 'sinon';
 import { mount } from 'enzyme';
 import { expect } from 'chai';
-import ConfirmModal from '~/components/ConfirmModal';
+import { ConfirmModalBody } from '~/components/modals';
 
 describe('components/modal/ConfirmModal', () => {
```
Comment (briansteffens) [inline test/components/ConfirmModal.spec.js]: Change to ConfirmModalBody ?

**Comment by eatonphil** [comment]
All the modals that I can see from a `grep -r showModal src`:

* [ ] Delete linode modal
* [ ] Delete DNS zone modal (not super important, but it does exist)
* [ ] Select configuration modal on boot / reboot modal
* [ ] Cancel backup service modal
* [ ] Reset root password modal
* [ ] Config deletion modal
* [ ] Edit disk modal
* [ ] Add disk modal
* [ ] Delete disk modal

**Pull request merged**